### PR TITLE
feat: Add upgrade/changelog code

### PR DESCRIPTION
**Pull request by andreifilip123**
Closes #<issue>

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

- Add a way for users to upgrade from an older version of ct3a to the latest

---

## Screenshots

image
image

### CODE REVIEW

File: upgrade/README.md
Comment (juliusmarminge) [inline upgrade/README.md:1]: better readme to tell folks what this is and how to use it

File: upgrade/.eslintrc.cjs
Comment (juliusmarminge) [inline upgrade/.eslintrc.cjs:1]: this should extend from the root eslint config

File: upgrade/next.config.mjs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  /**
+   * If you have `experimental: { appDir: true }` set, then you must comment the below `i18n` config
+   * out.
+   *
+   * @see https://github.com/vercel/next.js/issues/41980
+   */
+  i18n: {
+    locales: ["en"],
+    defaultLocale: "en",
+  },
```
Comment (juliusmarminge) [inline upgrade/next.config.mjs:20]: this uses appDir

```suggestion
```

File: upgrade/prettier.config.cjs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/** @type {import("prettier").Config} */
+const config = {
+  plugins: [require.resolve("prettier-plugin-tailwindcss")],
+};
```
Comment (juliusmarminge) [inline upgrade/prettier.config.cjs:4]: extend from root config

File: upgrade/src/lib/utils.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+};
+
+export type VersionsGroupedByMajor = Array<{
+  major: string;
+  versions: string[];
+}>;
+
+export const getT3Versions = async () => {
+  const response = await fetch(
+    "https://api.github.com/repos/t3-oss/create-t3-app/releases?per_page=100"
```
Comment (juliusmarminge) [inline upgrade/src/lib/utils.ts:19]: octokit?

File: upgrade/src/lib/utils.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    return "https://trpc.io/";
+  } else if (feature === "tailwind") {
+    return "https://tailwindcss.com/";
+  }
+};
+
+type VersionsRegex = {
+  currentVersion: string;
+  upgradeVersion: string;
+};
```
Comment (juliusmarminge) [inline upgrade/src/lib/utils.ts:91]: i recognize some of this code from the diff repo, forgot to remove somthing or is it needed in both?

Comment (andreifilip123) [reply upgrade/src/lib/utils.ts:109]: It's needed in both. I removed the unnecessary methods already

**Comment by andreifilip123** [comment]
https://t3-upgrade.vercel.app/
Here

**Pull request merged**